core/src/test/java/org/infinispan/expiration/impl/ClusterExpirationFunctionalTest.java

```diff
@@ -432,4 +439,32 @@ private void testMaxIdleNodeDies(Function<Cache<Object, String>, Object> keyToUs
 
       assertNotNull(cache1.get(key));
    }
+
+   public static final int LIFESPAN = 64;
+
+   public void testPrimaryNotExpiredButBackupWas() throws InterruptedException, ExecutionException, TimeoutException {
+      if (transactional || cacheMode != CacheMode.DIST_SYNC) {
```

Comment:
so, isn't the replicated cache affected by this?

Reply:
It is, but the test wasn't utilizing it :(

Reply:
I will try to tweak it to add it, requires slightly different rpc handling.

Reply:
REPL is broken atm, have to figure out how to fix it still.